### PR TITLE
Return the exact status code for P4Runtime request

### PR DESCRIPTION
**Pull request by nupuruttarwar**
Signed-off-by: Nupur Uttarwar <nupur.uttarwar@intel.com>

### CODE REVIEW

File: stratum/hal/lib/tdi/tdi_sde_table_entry.cc [hunk 1]
```diff
@@ -143,9 +147,14 @@ ::util::Status TdiSdeWrapper::DeleteTableEntry(
   const auto flags = ::tdi::Flags(0);
   tdi_status_t status = table->entryDel(
       *real_session->tdi_session_, *dev_tgt, flags, *real_table_key->table_key_);
-  if (!((status == BF_SUCCESS) || (status == BF_OBJECT_NOT_FOUND)))
+
+  if ((status == BF_OBJECT_NOT_FOUND)) {
+     return MAKE_ERROR(::util::error::Code::NOT_FOUND) <<
+             "No matching table entry with " << dump_args();
+  } else if (!(status == BF_SUCCESS)) {
```
Comment (ffoulkes) [inline stratum/hal/lib/tdi/tdi_sde_table_entry.cc]: `(status != BF_SUCCESS)` would be a clearer (and wouldn't require a second set of parentheses)

File: stratum/hal/lib/tdi/tdi_sde_table_entry.cc [hunk 2]
```diff
@@ -68,10 +68,14 @@ ::util::Status TdiSdeWrapper::InsertTableEntry(
   tdi_status_t status = table->entryAdd(
       *real_session->tdi_session_, *dev_tgt, flags, *real_table_key->table_key_,
       *real_table_data->table_data_);
-  if (!((status == BF_SUCCESS) || (status == BF_ALREADY_EXISTS)))
+  if ((status == BF_ALREADY_EXISTS)) {
+     return MAKE_ERROR(::util::error::Code::ALREADY_EXISTS) <<
+             "Duplicate table entry with " << dump_args();
+  } else if (!(status == BF_SUCCESS)) {
```
Comment (ffoulkes) [inline stratum/hal/lib/tdi/tdi_sde_table_entry.cc]: `(status != BF_SUCCESS)` would be clearer (and wouldn't require a second set of parentheses)

File: stratum/hal/lib/tdi/tdi_sde_table_entry.cc [hunk 3]
```diff
@@ -68,10 +68,14 @@ ::util::Status TdiSdeWrapper::InsertTableEntry(
   tdi_status_t status = table->entryAdd(
       *real_session->tdi_session_, *dev_tgt, flags, *real_table_key->table_key_,
       *real_table_data->table_data_);
-  if (!((status == BF_SUCCESS) || (status == BF_ALREADY_EXISTS)))
+  if ((status == BF_ALREADY_EXISTS)) {
```
Comment (ffoulkes) [inline stratum/hal/lib/tdi/tdi_sde_table_entry.cc]: this only needs one pair of parentheses

File: stratum/hal/lib/tdi/tdi_sde_table_entry.cc [hunk 4]
```diff
@@ -143,9 +147,14 @@ ::util::Status TdiSdeWrapper::DeleteTableEntry(
   const auto flags = ::tdi::Flags(0);
   tdi_status_t status = table->entryDel(
       *real_session->tdi_session_, *dev_tgt, flags, *real_table_key->table_key_);
-  if (!((status == BF_SUCCESS) || (status == BF_OBJECT_NOT_FOUND)))
+
+  if ((status == BF_OBJECT_NOT_FOUND)) {
```
Comment (ffoulkes) [inline stratum/hal/lib/tdi/tdi_sde_table_entry.cc]: this only needs one pair of parentheses

**Review by ffoulkes — APPROVED** [review]
One of the quality recommendations for code reviews is to "write good change descriptions."

Since you're changing the behavior of the method, it would be useful to the maintainer to know why. Is it to allow a higher-level to to make a decision based on the status code? To provide a better description for a log message?

One way of doing this is to provide a problem statement, followed by a statement of what you're changing (and an explanation if the connection between the two isn't obvious).

Otherwise, LGTM

File: stratum/hal/lib/tdi/tdi_sde_table_entry.cc (hunk 3 above)
Comment (nupuruttarwar) [reply stratum/hal/lib/tdi/tdi_sde_table_entry.cc]: Done

File: stratum/hal/lib/tdi/tdi_sde_table_entry.cc (hunk 2 above)
Comment (nupuruttarwar) [reply stratum/hal/lib/tdi/tdi_sde_table_entry.cc]: Done

File: stratum/hal/lib/tdi/tdi_sde_table_entry.cc (hunk 4 above)
Comment (nupuruttarwar) [reply stratum/hal/lib/tdi/tdi_sde_table_entry.cc]: Done

File: stratum/hal/lib/tdi/tdi_sde_table_entry.cc (hunk 1 above)
Comment (nupuruttarwar) [reply stratum/hal/lib/tdi/tdi_sde_table_entry.cc]: Done

**Comment by nupuruttarwar** [comment]
> One of the quality recommendations for code reviews is to "write good change descriptions."
> 
> Since you're changing the behavior of the method, it would be useful to the maintainer to know why. Is it to allow a higher-level to to make a decision based on the status code? To provide a better description for a log message?
> 
> One way of doing this is to provide a problem statement, followed by a statement of what you're changing (and an explanation if the connection between the two isn't obvious).
> 
> Otherwise, LGTM

Modified the commit messgae

**Pull request merged**